### ISSUE TITLE
a consistent api for when you're running from an IDE

### ISSUE THREAD

**Issue by xenoterracide** [gradle/gradle#17275]
I'd like an api that allows me to detect when I'm in an IDE, I think that some like Intellij do set a property or env var or something, but I also think this is a proprietary property, meaning I'd have to check for intellij, netbeans, eclipse, etc.

### Expected Behavior

```kts
if ( gradleRuntime.isIde ) {
    // do stuff
}
```
### Context

some plugins, specifically compiler plugins, I'd like to customize based on whether they're running in an IDE. for example error prone has a check for detecting unused variables, this is generally great, but if I'm running a debugger it's annoying because sometimes I add a variable to look at it in the debugger but never actually use it, for example when I make an API call normally as void ignoring the return.

**Comment by bmuskalla** [comment]
We have no intention of owning this decision. There are too many different aspects what "is inside IDE" or not. Also this would interfere with daemon lifecycle, caching, etc. As you said, there are ways to detect this and should be aligned with your usecase.

**Comment by xenoterracide** [comment]
yeah, but as I said, there is not a consistent way for an IDE to tell gradle so that you can check, so you'd have to check each one, which means researching something like 20 different IDE's editors.

**Comment by xenoterracide** [comment]
I don't expect gradle to provide the information, honestly I'm thinking it could easily be more of a wrapper API around a property like idea sets, or it could even just be a specification of a property for the IDE to set so it's consistent among IDE's (which requires only documenting it).
